src/rdkafka.c

```diff
@@ -2127,7 +2127,12 @@ static int rd_kafka_thread_main(void *arg) {
                                        RD_KAFKA_CGRP_STATE_TERM)))) {
                 rd_ts_t sleeptime = rd_kafka_timers_next(
                     &rk->rk_timers, 1000 * 1000 /*1s*/, 1 /*lock*/);
-                rd_kafka_q_serve(rk->rk_ops, (int)(sleeptime / 1000), 0,
+                /* Round up to avoid calling serve with timeout 0 ms
+                 * in a tight loop until 1 ms has passed. */
+                int timeout_ms = (int)(sleeptime / 1000);
+                if (sleeptime % 1000 > 0)
+                        timeout_ms++;
```

Comment:
This is the same as
```
int timeout_ms = (int)(sleeptime / 1000) + (sleeptime % 1000 > 0);
```
or perhaps even better,
```
int timeout_ms = ((sleeptime + 999) / 1000);
```
(The cast to `int` probably isn't needed, unless it's suppressing a compiler warning.)

Also, this diff seems totally unrelated to the "metadata cache" stuff in this PR. This should be a separate commit with a separate PR.

Reply:
Thanks, I think I'll use the second one

Reply:
Putting it in a different commit too